lib/globalize/active_record/migration.rb

```diff
@@ -166,7 +166,16 @@ def validate_translated_fields
         end
 
         def column_type(name)
-          columns.detect { |c| c.name == name.to_s }.try(:type) || :string
+          column_descriptor = columns.detect { |c| c.name == name.to_s }
+          column_descriptor_type = column_descriptor.try(:type)
+          column_descriptor_sql_type = column_descriptor.try(:sql_type)
+
+          # return :bigint for the type instead of :integer
```

Comment:
could you write _why_ this is happening please?

Reply:
You mean, why typing misconfiguration happening? In a nutshell, before this PR globalize passed a type of `td.reference` during the creation of the translation table, which it took from `column_definition.type` ("Rails"-type), instead of a real data type. Here is an example of those in AR:
https://github.com/rails/rails/blob/b2eb1d1c55a59fee1e6c4cba7030d8ceb524267c/activerecord/test/cases/adapters/postgresql/serial_test.rb#L65-L70

I didn't find any proper documentation or comments why exactly AR do split those types and wouldn't return `:bigint` on `cd.type` call instead of `:integer`.
Should I put anything to the comment in the code? Wouldn't like to blow it up with this whole story. We could simply put a link to this thread. What do you think?

Reply:
yes, a link to the pull request at least would be really helpful when trying to figure this code out in the years to come, if we're still doing this :smile: 	thank you!